### PR TITLE
Fix indicatif prog bar

### PR DESCRIPTION
**Pull request by tgauth**
# PR Summary
- readdress #337 
- uses `warn_span` for progress bars since that is the default tracing level and can be turned off by specifying `error` trace level
- no longer displays progress bars once progress is completed, so `done` messages are removed

## PR Context

- `dsc resource list` demo:

https://github.com/PowerShell/DSC/assets/14894321/c98ad0d2-6581-4f88-a593-f3379132317b

- `dsc config get` demo:

https://github.com/PowerShell/DSC/assets/14894321/82948b41-dc58-45f8-aee5-798182ee04d2

### CODE REVIEW

File: dsc_lib/src/configure/mod.rs [hunk 1]
```diff
@@ -135,13 +135,13 @@ fn escape_property_values(properties: &Map<String, Value>) -> Result<Option<Map<
     Ok(Some(result))
 }
 
-fn get_progress_bar(len: u64) -> Result<ProgressBar, DscError> {
-    let pb = ProgressBar::new(len);
-    pb.enable_steady_tick(Duration::from_millis(120));
-    pb.set_style(ProgressStyle::with_template(
+fn get_progress_bar_span(len: u64) -> Result<Span, DscError> {
+    let pb_span = warn_span!("progress bar");
```
Comment (SteveL-MSFT) [inline dsc_lib/src/configure/mod.rs:139]: If we just have this be an empty string `warn_span!("")` would that suppress the unnecessary `progress bar` in the traces?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: dsc_lib/src/configure/mod.rs [hunk 2]
```diff
@@ -246,7 +247,8 @@ impl Configurator {
             result.results.push(resource_result);
         }
 
-        pb.finish_with_message("Get configuration completed");
+        mem::drop(pb_span_enter);
```
Comment (SteveL-MSFT) [inline dsc_lib/src/configure/mod.rs:250]: is this needed explicitly?  I would expect both to be dropped when going out of scope

File: dsc_lib/src/configure/mod.rs [hunk 3]
```diff
@@ -135,13 +135,13 @@ fn escape_property_values(properties: &Map<String, Value>) -> Result<Option<Map<
     Ok(Some(result))
 }
 
-fn get_progress_bar(len: u64) -> Result<ProgressBar, DscError> {
-    let pb = ProgressBar::new(len);
-    pb.enable_steady_tick(Duration::from_millis(120));
-    pb.set_style(ProgressStyle::with_template(
+fn get_progress_bar_span(len: u64) -> Result<Span, DscError> {
+    let pb_span = warn_span!("");
```
Comment (SteveL-MSFT) [inline dsc_lib/src/configure/mod.rs:139]: Presumably using a warn span means that if tracing level is error, then progress isn't displayed right?  That seems fine but would need to be documented.

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: dsc_lib/src/configure/mod.rs (hunk 2 above)
Comment (tgauth) [reply dsc_lib/src/configure/mod.rs]: it's needed within the config functions but I removed it from resource list

File: dsc_lib/src/configure/mod.rs (hunk 3 above)
Comment (tgauth) [reply dsc_lib/src/configure/mod.rs:140]: Correct - added a comment. Wasn't sure where to call that out initially, so it was just in the PR description.

**Review by SteveL-MSFT — APPROVED** [review]

**Pull request merged**